extensions/amp-story/1.0/animation.js

```diff
@@ -152,6 +159,33 @@ class AnimationRunner {
     this.runnerPromise_.then((runner) => this.onRunnerReady_(runner));
   }
 
+  /**
+   * @param {!Element} page
+   * @param {!StoryAnimationDef} animationDef
+   * @param {!Promise<!../../amp-animation/0.1/web-animations.Builder>} webAnimationBuilderPromise
+   * @param {!../../../src/service/vsync-impl.Vsync} vsync
+   * @param {!AnimationSequence} sequence
+   * @param {!Object<string, *>=} keyframeOptions
+   * @return {!AnimationRunner}
+   */
+  static create(
+    page,
+    animationDef,
+    webAnimationBuilderPromise,
+    vsync,
+    sequence,
+    keyframeOptions = {}
+  ) {
+    return new AnimationRunner(
+      page,
+      animationDef,
+      webAnimationBuilderPromise,
+      vsync,
+      sequence,
+      keyframeOptions
+    );
+  }
+
```

Comment:
What's the need for this block? Why not call `new AnimationRunner` directly?

Reply:
It needs to be spied/stubbed. Because of how we transpile, we can't mock imported classes, but we _can_ stub their static methods.

Reply:
As much as possible we should not fake the classes, especially this one that is part of the code we're testing. How hard would it be to not fake it in the tests?

Reply:
I'm faking the class to ensure that it's instantiated properly. The class itself is tested in isolation.

This is a unit test, not an e2e test ;)

Reply:
(This is also the only wait to test that the correct preset is passed, since its keyframes are evaluated (and therefore different than a comparison object) before being passed to the amp-animation's `WebAnimationRunner`)

Reply:
why can't we use the actual class while ensuring it's instantiated properly? in theory we could still spy the `animationDef` and ensure it's passing the right preset, no?

Reply:
That would complicate things:

1. This would mean testing the flow from the tree (instantiating `AnimationManager` only) to the `amp-animation` interfaces. To test all the current cases, we'd introduce an amount of tests roughly equal to `AnimationManager cases * AnimationRunner cases * AnimationSequence cases`. 

2. You can't compare presets directly at this level. You have to reconstruct them from internal `AnimationRunner` logic and their keyframe functions. So you either hardcode all constructed values for all tested configurations, or you duplicate `AnimationRunner` logic in tests, neither of which is great.